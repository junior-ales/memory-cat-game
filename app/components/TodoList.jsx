import React from 'react';
import { compose, map } from 'ramda';
import { deleteTodo } from '../actions';
import TodoItem from './TodoItem';

const TodoList = ({todos, onDelete}) => {
  return (
    <ul>
      {map(todo => TodoItem({todo, onDelete}), todos)}
    </ul>
  );
};

TodoList.propTypes = { // eslint-disable-line
  todos: React.PropTypes.array.isRequired,
  onDelete: React.PropTypes.func.isRequired
};

const extractProps = props => ({
  todos: props.todos,
  onDelete: props.dispatch(deleteTodo)
});

export default compose(TodoList, extractProps);
