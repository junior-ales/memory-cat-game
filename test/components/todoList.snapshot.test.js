/* eslint-disable */
import renderer from 'react-test-renderer';
import React from 'react';
import TodoList from '../../app/components/TodoList';

const props = {
  todos: [
    { id: 1, text: 'todo1', completed: false },
    { id: 2, text: 'todo2', completed: false },
    { id: 3, text: 'todo3', completed: false }
  ],
  dispatch: action => id => action(id)
};

test('should render todo list', () => {
  const todoList = renderer.create(<TodoList {...props} />);
  expect(todoList.toJSON()).toMatchSnapshot();
});
