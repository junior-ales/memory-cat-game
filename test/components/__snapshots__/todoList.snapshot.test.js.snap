exports[`test should render todo list 1`] = `
<ul>
  <li>
    todo1
    <button
      onClick={[Function]}>
      Remove
    </button>
  </li>
  <li>
    todo2
    <button
      onClick={[Function]}>
      Remove
    </button>
  </li>
  <li>
    todo3
    <button
      onClick={[Function]}>
      Remove
    </button>
  </li>
</ul>
`;
